include/cpptoml.h

```diff
@@ -307,9 +323,27 @@ class value : public base
 
   private:
     T data_;
+
+    /**
+     * Constructs a value from the given data.
+     */
+    value(const T& val) : data_(val)
+    {
+    }
+
+    value(const value& val) = delete;
+    value& operator=(const value& val) = delete;
 };
 
 template <class T>
+inline std::shared_ptr<typename value_traits<T>::type> make_value(T&& val)
+{
+    typedef std::shared_ptr<typename value_traits<T>::type> pointer_type;
+    return pointer_type(
+        new typename pointer_type::element_type(std::forward<T>(val)));
+}
```

Comment:
Very crafty.  I played with it some to try to get make_shared to work, but I attacked it from friending the right function.  Unfortunately, this turns out to be a slightly different function on every compiler.